### DISCUSSION TITLE
Postal stops sending emails

### DISCUSSION THREAD

**Discussion by ndTEC** [postalserver/postal#1647] [Help with using Postal]
As of #1643 I am coming back with more information @willpower232 
Today it stopped sending emails. We send postal logs to kibana so i took a look to find anything.
In rabbitmq i noticed the queue messages going up. In the logs from postal during the time it didn't work i couldn't find "Got queued message with exclusive lock" or "Started processing RequeueWebhooksJob job" which are plenty otherwise. All containers were online and i could docker exec inside them. All this time the queued messages in Postal was increasing fast. 
This is not the first time it happened since we migrated to v2 two weeks ago.

**Comment by ndTEC** [comment]
It happened again. Found this in worker logs. Rabbitmq is installed in container.
ERROR -- #<Bunny::Session:0x55aac927dd40 postal@127.0.0.1:5672, vhost=/postal, addresses=[127.0.0.1:5672]>: Uncaught exception from consumer #<Bunny::Consumer:47096004028560 @channel_id=1 @queue=deliver-jobs-main> @consumer_tag=bunny-1634624125000-514812760234>: #<Bunny::ChannelAlreadyClosed: cannot use a closed channel! Channel id: 1, closed due to a server-reported channel error: PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more> @ /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/channel.rb:1999:in `raise_if_no_longer_open!'

And in rabbitmq logs i have:
Channel error on connection <0.9782.2> (172.18.0.1:60832 -> 172.18.0.2:5672, vhost: '/postal', user: 'postal'), channel 1:
operation none caused a channel exception precondition_failed: delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more
![image_2021_10_20T12_01_15_973Z](https://user-images.githubusercontent.com/79922948/138090478-13599dcc-e2d4-40bc-9103-edc8e193acb3.png)

It seems the worker an rabbitmq loses conection.

**Reply by willpower232** [reply]
Interesting, I've never seen that before. How many messages are you trying to send?

**Comment by ndTEC** [comment]
During the day we have our peak at aprox 130/minute, but on average i would say 15-20/minute.
In the screenshot with rabbitmq at Consumers it should be the number of workers, but now even though i have 4 workers it says 3 Consumers. After checking logs for each worker i found the same error for postal_worker_3

ERROR -- #<Bunny::Session:0x561503564d98 postal@127.0.0.1:5672, vhost=/postal, addresses=[127.0.0.1:5672]>: Uncaught exception from consumer #<Bunny::Consumer:47324125412820 @channel_id=1 @queue=deliver-jobs-main> @consumer_tag=bunny-1634737093000-101046368085>: #<Bunny::ChannelAlreadyClosed: cannot use a closed channel! Channel id: 1, closed due to a server-reported channel error: PRECONDITION_FAILED - delivery acknowledgement on channel 1 timed out. Timeout value used: 1800000 ms. This timeout value can be configured, see consumers doc guide to learn more> @ /usr/local/bundle/gems/bunny-2.14.4/lib/bunny/channel.rb:1999:in `raise_if_no_longer_open!'

After restarting postal_worker_3 now i have again 4 Consumers. So i think there is a problem even with rabbitmq or most probably with postal workers because they disconnect from rabbitmq.

**Reply by willpower232** [reply]
it is possible this is a thing with newer rabbitmq versions apparently https://stackoverflow.com/questions/68929753/rabbitmq-delivery-acknowledgement-timeout-workarounds-pika-or-editing-rabbitmq

**Comment by ndTEC** [comment]
In what circumstances does Postal need more than 30 minutes to process an email? I should have a large amount of emails coming at Postal and not enough resources in order to take more than 30 minutes to process an email and reach the 30 minutes timeout, but it's not my case. Anyway in rabbitmq i rarely have spikes 3000-4000 messages in queue, for 2-3 minutes and then it's ok. Let's suppose a message reaches the timeout period, is it normal for that worker to permanently disconnects from rabbitmq?

**Reply by willpower232** [reply]
Its likely that an existing connection is reused but needs refreshing at a certain time so if the configuration of rabbitmq and the worker is wildly different then this might be a problem but I don't know enough about rabbitmq.
